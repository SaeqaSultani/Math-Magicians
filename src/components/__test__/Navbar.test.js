import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../navbar';

const MockTodoFooter = () => (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

it('Check for any changes to the component', () => {
  const tree = renderer.create(
    <MockTodoFooter />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Three links should be rendered', () => {
  render(
    <MockTodoFooter />,
  );
  const listItems = screen.getAllByTestId('li');

  expect(listItems.length).toBe(3);
});
