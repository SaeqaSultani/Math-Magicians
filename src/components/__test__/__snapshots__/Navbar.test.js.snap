// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check for any changes to the component 1`] = `
<div
  className="container"
  data-testid="navbar"
>
  <h1>
    Math Magicians
  </h1>
  <div
    className="navbar"
  >
    <div
      className="nav-item"
      data-testid="li"
    >
      <a
        href="/"
        onClick={[Function]}
      >
         Home
      </a>
    </div>
    <span
      className="nav-item"
    >
      |
    </span>
    <div
      className="nav-item"
      data-testid="li"
    >
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </div>
    <span
      className="nav-item"
    >
      |
    </span>
    <div
      className="nav-item"
      data-testid="li"
    >
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </div>
</div>
`;
